File: apps/restaurant_app/lib/core/rbac/permissions.dart
/// Mirrors `PERMISSIONS` in `services/api/src/common/rbac/permissions.catalog.ts`. Keep the
/// two lists in sync by hand for now — v1 has no shared-types codegen pipeline for Dart (see
/// docs/architecture.md §13; `packages/shared_types` only serves the TypeScript customer web
/// app so far).
///
/// IMPORTANT: this list is for *client-side UI gating only* — hiding a "Void order" button
/// from a Waiter makes for a better app, but it is not what stops a Waiter from voiding an
/// order. That enforcement is `PermissionsGuard` on the server, on every request, unconditionally.
/// Never treat a check against this list as a security boundary.
class Permissions {
  const Permissions._();

  static const ordersCreate = 'orders.create';
  static const ordersUpdate = 'orders.update';
  static const ordersCancel = 'orders.cancel';
  static const ordersDiscount = 'orders.discount';
  static const ordersView = 'orders.view';

  static const kitchenView = 'kitchen.view';
  static const kitchenUpdate = 'kitchen.update';

  static const billingCreate = 'billing.create';
  static const billingView = 'billing.view';

  static const paymentsTake = 'payments.take';
  static const paymentsRefund = 'payments.refund';
  static const paymentsView = 'payments.view';

  static const tablesManage = 'tables.manage';
  static const tablesView = 'tables.view';

  static const menuEdit = 'menu.edit';
  static const menuView = 'menu.view';

  static const reportsView = 'reports.view';

  static const staffManage = 'staff.manage';
  static const staffView = 'staff.view';

  static const settingsManage = 'settings.manage';

  static const auditView = 'audit.view';

  static const printersManage = 'printers.manage';
}

File: packages/shared_types/src/permissions.ts
/**
 * Mirrors `PERMISSIONS` in `services/api/src/common/rbac/permissions.catalog.ts` — the same
 * fixed, global permission-key catalog, for any TypeScript staff-facing frontend to gate UI
 * with (e.g. `apps/pos_web` hiding "Generate bill" from a user without `BILLING_CREATE`).
 * `apps/restaurant_app/lib/core/rbac/permissions.dart` keeps its own hand-mirrored copy for
 * the same reason `lib/features/pos/data/pos_models.dart` does — Dart can't import this
 * package (see this package's index.ts doc comment) — so all three lists (backend, Dart, this
 * one) must be kept in sync by hand.
 *
 * IMPORTANT: exactly like the Dart copy, this is for *client-side UI gating only*. Hiding a
 * button from a user without the permission makes for a better app; it is not what stops them
 * from calling the endpoint directly. That enforcement is `PermissionsGuard` on the server, on
 * every request, unconditionally — never treat a check against this list as a security boundary.
 */
export const PERMISSIONS = {
  ORDERS_CREATE: 'orders.create',
  ORDERS_UPDATE: 'orders.update',
  ORDERS_CANCEL: 'orders.cancel',
  ORDERS_DISCOUNT: 'orders.discount',
  ORDERS_VIEW: 'orders.view',

  KITCHEN_VIEW: 'kitchen.view',
  KITCHEN_UPDATE: 'kitchen.update',

  BILLING_CREATE: 'billing.create',
  BILLING_VIEW: 'billing.view',

  PAYMENTS_TAKE: 'payments.take',
  PAYMENTS_REFUND: 'payments.refund',
  PAYMENTS_VIEW: 'payments.view',

  TABLES_MANAGE: 'tables.manage',
  TABLES_VIEW: 'tables.view',

  MENU_EDIT: 'menu.edit',
  MENU_VIEW: 'menu.view',

  REPORTS_VIEW: 'reports.view',

  STAFF_MANAGE: 'staff.manage',
  STAFF_VIEW: 'staff.view',

  SETTINGS_MANAGE: 'settings.manage',

  AUDIT_VIEW: 'audit.view',

  PRINTERS_MANAGE: 'printers.manage',
} as const;

export type PermissionKey = (typeof PERMISSIONS)[keyof typeof PERMISSIONS];
